src/coordination/coordinator_state_manager.cpp

```diff
@@ -31,37 +33,59 @@ CoordinatorStateManager::CoordinatorStateManager(CoordinatorInstanceInitConfig c
 
   cluster_config_ = cs_new<cluster_config>();
   cluster_config_->get_servers().push_back(my_srv_config_);
+  utils::EnsureDirOrDie(config.durability_dir);
+  kv_store_ = std::make_unique<kvstore::KVStore>(config.durability_dir);
 }
 
 auto CoordinatorStateManager::load_config() -> ptr<cluster_config> {
-  // Just return in-memory data in this example.
-  // May require reading from disk here, if it has been written to disk.
-  spdlog::trace("Loading cluster config");
+  MG_ASSERT(kv_store_, "Durability folder should be created.");
+  spdlog::trace("Loading cluster config from RocksDb");
+  auto servers = kv_store_->Get("servers");
+  if (!servers.has_value()) {
+    spdlog::trace("Didn't find anything stored on disk for cluster config.");
+    return cluster_config_;
+  }
+  spdlog::trace("Loading cluster config from disk.");
+  auto const json = nlohmann::json::parse(servers.value());
+  auto const real_servers = json.get<std::vector<std::tuple<int, std::string, std::string>>>();
+  cluster_config_->get_servers().clear();
+  for (auto const &real_server : real_servers) {
+    auto const &[coord_id, endpoint, aux] = real_server;
+    spdlog::trace("Recreating cluster config with id: {}, endpoint: {} and aux data: {} from disk.", coord_id, endpoint,
+                  aux);
+    auto one_server_config = cs_new<srv_config>(coord_id, 0, endpoint, aux, false);
+    cluster_config_->get_servers().push_back(std::move(one_server_config));
+  }
+
   return cluster_config_;
 }
 
 auto CoordinatorStateManager::save_config(cluster_config const &config) -> void {
-  // Just keep in memory in this example.
-  // Need to write to disk here, if want to make it durable.
+  MG_ASSERT(kv_store_, "Disk folder should be created");
   ptr<buffer> buf = config.serialize();
   cluster_config_ = cluster_config::deserialize(*buf);
-  spdlog::info("Saving cluster config.");
+  spdlog::info("Saving cluster config to disk.");
   auto servers = cluster_config_->get_servers();
+  std::vector<std::tuple<int, std::string, std::string>> servers_vec;
```

Comment:
```suggestion
auto const servers_vec = ranges::views::transform(cluser_config_->get_servers(), [](auto const &server) {
  return std::make_tuple(...);
}) | to<std::vector>();
```

No need to change but with this you avoid some boring half bugs like missing const on servers or specifying whole std::tuple ...

Reply:
Ah, ranges, nice!